core/node/sync_job.cpp

```diff
@@ -268,24 +268,30 @@ namespace fc::sync {
     if (branch == last) {
       return;
     }
+    TsBranchPtr prev_branch;
```

Comment:
prev_branch using like a flag. Can we use prev_it instead of it?

Does it needed to save TsBranch?

Reply:
`prev_branch` is `shared_ptr` to `map` owning `it`, so we need to keep `prev_branch` alive.